src/Utils/hiopOptions.cpp

```diff
@@ -849,9 +514,614 @@ void hiopOptions::OptionInt::print(FILE* f) const
+    }
+    set_val("compute_mode", "cpu");
+  }
+  if(GetString("compute_mode")=="gpu") {
+    log_printf(hovWarning,
+               "option compute_mode=gpu was changed to 'cpu' since HiOp was built without GPU support.\n");
+    set_val("compute_mode", "cpu");
+  }
+  
+  if(GetString("compute_mode")=="auto") {
+    set_val("compute_mode", "cpu");
+  }
+#endif
+  
+  // No removing of fixed variables in GPU compute mode ...
+  if(GetString("compute_mode")=="gpu") {
+    if(GetString("fixed_var")=="remove") {
+      
+      log_printf(hovWarning,
+                 "option fixed_var=remove was changed to 'relax' since only 'relax'"
+                 "is supported in GPU compute mode.\n");
+      set_val("fixed_var", "relax");
+    }
+  }
+
+}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// hiopOptionsPriDec
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+hiopOptionsPriDec::hiopOptionsPriDec(const char* opt_filename/*=nullptr*/)
+  : hiopOptions()
+{
+  register_options();
+  load_from_file(opt_filename==nullptr ? hiopOptions::default_filename_pridec_solver : opt_filename);
+  ensure_consistence();
+}
+hiopOptionsPriDec::~hiopOptionsPriDec()
+{
+}
+
+void hiopOptionsPriDec::register_options()
+{
+  //
+  // Primal decomposition (PriDec) solver
+  //
+  
+  //name of the options file to be passed to the master solver (by the NLP solver, e.g., HiOp or Ipopt or other)
+  {
+    register_str_option("options_file_master_prob",
+                        hiopOptions::default_filename_pridec_masterNLP,
+                        "Options file for the NLP solver solving the master problem in PriDec solver");
+  }
+
+  //
+  // portability
+  //
+  {
+#ifdef HIOP_USE_RAJA
+    vector<string> range(4);
+    range[0] = "default";
+    range[1] = "host";
+    range[2] = "device";
+    range[3] = "um";
+#else
+    vector<string> range(1);
+    range[0] = "default";
+#endif
+    register_str_option("mem_space",
+                        range[0],
+                        range,
+                        "Determines the memory space used by PriDec solver for linear algebra objects. Must match the "
+                        "the memory space in which the master solve is going to be done.");
+  }
+  
+  //
+  // convergence and stopping criteria
+  //
+  {
+    //TODO: Frank check these and add others as needed in the primal decomposition algorithm
+    register_num_option("tolerance",
+                        1e-4,
+                        1e-14,
+                        1e-1,
+                        "Absolute error tolerance for the PriDec solver (default 1e-4)");
+    
+    //register_num_option("rel_tolerance", 0., 0., 0.1,
+    //                  "Error tolerance for the NLP relative to errors at the initial point. A null "
+    //                  "value disables this option (default 0.)");
```

Comment:
Should we remove these 3 lines from the source code?

Reply:
these are instructions for Frank's iteration over the PriDec solver to support new options.